solver/jobs.go

```diff
@@ -495,17 +496,44 @@ func (jl *Solver) deleteIfUnreferenced(k digest.Digest, st *state) {
 	}
 }
 
-func (j *Job) Build(ctx context.Context, e Edge) (CachedResult, error) {
+func (j *Job) Build(ctx context.Context, e Edge) (CachedResult, BuildInfo, error) {
 	if span := trace.SpanFromContext(ctx); span.SpanContext().IsValid() {
 		j.span = span
 	}
 
 	v, err := j.list.load(e.Vertex, nil, j)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	e.Vertex = v
-	return j.list.s.build(ctx, e)
+
+	res, err := j.list.s.build(ctx, e)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return res, j.walkBuildInfo(ctx, e), nil
+}
+
+func (j *Job) walkBuildInfo(ctx context.Context, e Edge) BuildInfo {
+	j.list.mu.Lock()
+	defer j.list.mu.Unlock()
+	bi := make(BuildInfo)
+	// FIXME: Walking should start from e
+	for _, st := range j.list.actives {
```

Comment:
@tonistiigi Still can't figure out how to properly walk from `e` and find the matching state from `j.list.actives`. Do you have an idea?

Reply:
Post your attempt. I'll take a look at what's wrong.

The question in here is that if the client does `Build()` and multiple `Solve` requests in the callback with the last one returning a result, should buildinfo be only for the last solve or for all? I tend to think the former. If for all then the current one is correct.